Library/Homebrew/cmd/install.rb

```diff
@@ -14,16 +14,10 @@ def install
     end
 
     ARGV.named.each do |name|
-      # if a formula has been tapped ignore the blacklisting
-      unless Formula.path(name).file?
-        msg = blacklisted? name
-        raise "No available formula for #{name}\n#{msg}" if msg
+      if name =~ HOMEBREW_TAP_FORMULA_REGEX || name =~ HOMEBREW_CASK_TAP_FORMULA_REGEX
+        install_tap $1, $2 unless (HOMEBREW_LIBRARY/"Taps/#{$1.downcase}/homebrew-#{$2.downcase}").directory?
       end
-      if !File.exist?(name) && (name =~ HOMEBREW_TAP_FORMULA_REGEX \
-                                || name =~ HOMEBREW_CASK_TAP_FORMULA_REGEX)
```

Comment:
I'm pretty sure it had something to do with installing from a path (e.g. `brew install foo/bar/baz.rb`) matching the tap regex.